aten/src/ATen/native/mps/kernels/ActivationKernel.metal

```diff
@@ -132,7 +127,9 @@ struct leaky_relu_backward_functor {
       const T self,
       const T grad_output,
       const T negative_slope) {
-    return self > T(0) ? grad_output : grad_output * negative_slope;
+    return float(self) > 0.0f
```

Comment:
Just curious if there are every a situation when compiler will not be able to infer types here
```suggestion
    return self > 0.
```